records_mover/url/optimizer/__init__.py

```diff
@@ -0,0 +1,232 @@
+        # TODO
+        current_year = 2020
+        # TODO
+        sink_bucket = "bluelabs-test-recordsmover"
+        source_bucket = loc.bucket
+        aws_creds = loc.aws_creds()
+        if aws_creds is None:
+            logger.warning("S3 bucket did not provide AWS creds - "
+                           "cannot copy S3 bucket using Google Storage Transfer.  "
+                           "Falling back to a slower method of bucket copy.")
+            return False
+        if aws_creds.token is not None:
+            logger.warning("S3 bucket is using a temporary access token (MFA creds?) which "
+                           "Google Storage Transfer does not support.  Falling back to a slower "
+                           "method of bucket copy.")
+            return False
+
+        access_key_id = aws_creds.access_key
+        secret_access_key = aws_creds.secret_key
+
+        # Edit this template with desired parameters.
+        transfer_job: '_TransferJobConfig' = {
+            'description': description,
+            'status': 'ENABLED',
+            'projectId': project_id,
+            'schedule': {
+                'scheduleStartDate': {
+                    'day': current_day,
+                    'month': current_month,
+                    'year': current_year,
+                },
+                'scheduleEndDate': {
+                    'day': current_day,
+                    'month': current_month,
+                    'year': current_year,
+                }
+            },
+            'transferSpec': {
+                'awsS3DataSource': {
+                    'bucketName': source_bucket,
+                    'awsAccessKey': {
+                        'accessKeyId': access_key_id,
+                        'secretAccessKey': secret_access_key
+                    }
+                },
+                "objectConditions": {
+                    "includePrefixes": [
+                        loc.key
+                    ]
+                },
+                "transferOptions": {
+                    "overwriteObjectsAlreadyExistingInSink": False,
+                    "deleteObjectsUniqueInSink": False,
+                    "deleteObjectsFromSourceAfterTransfer": False
+                },
+                'gcsDataSink': {
+                    'bucketName': sink_bucket
+                }
+            }
+        }
+
+        result = storagetransfer.transferJobs().create(body=transfer_job).execute()
+        print('Returned transferJob: {}'.format(
+            json.dumps(result, indent=4)))
+        job_name = result['name']
+        self._wait_for_transfer_job(project_id, job_name, gcp_credentials)
+        return True
+
+    def _wait_for_transfer_job(self,
+                               project_id: str,
+                               job_name: str,
+                               gcp_credentials: 'google.auth.credentials.Credentials') -> None:
+        import googleapiclient
+
+        # TODO: Pass in correct creds
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker